### PR TITLE
STM32F1 support

### PR DESCRIPTION
**Pull request by blazewicz**
Both blink and usart examples were tested on STM32F103C8T6.

Clock configuration was verified with oscilloscope.

I found this project yesterday and I am not very familiar with API, so any suggestion are welcome.

### CODE REVIEW

File: src/hal/stm32f1/iomem.ld [hunk 1]
```diff
@@ -0,0 +1,37 @@
+stm32f1_iomem_PWR   = 0x40007000;
```
Comment (farcaller) [inline src/hal/stm32f1/iomem.ld]: It looks like stm32f1 has proper svd files: https://github.com/posborne/cmsis-svd/tree/master/data/STMicro, that should be the preferred method to generate ioregs now, as `extern` registers and `iomem.ld` is now deprecated, replaced by placement ioregs.

**Comment by andrey-gvrd** [comment]
@blazewicz I'm interested in working on this, can I help?

**Comment by blazewicz** [comment]
@andrey-gvrd that's great! I appreciate any help

**Comment by andrey-gvrd** [comment]
@blazewicz @farcaller 

Wanted to ask about a strategy of getting this resolved before going off too far in a wrong direction. I see two options:

1. Get this PR compiling on a version of `rustc` that this PR originally used and then port it to current version of  compiler and `zinc`;
2. Cherry-pick this PR's commits one-by-one and get them compiling with current compiler and `zinc`.

Thoughts?

**Comment by farcaller** [comment]
The latter sounds better. The problem is that ioreg became a pretty big showstopper, as it breaks every now and then, and I can't dedicate enough time to clean this mess up for just too long now.

The best strategy would be to generate ioregs statically for this mcu and move forward knowing it doesn't depend on random breakage.

**Comment by andrey-gvrd** [comment]
Actually both examples for STM32F1 build just fine, this should probably be just merged in.

**Comment by blazewicz** [comment]
@farcaller what do you think? Do you want me to squash it into a single
commit?

**Comment by goandylok** [comment]
I have simply copied the stm32f1 definition to the newest master branch, and downloaded a blink example to my broard. Actually, it seemed to work well. Why not merge it?
p60705-232734

**Comment by farcaller** [comment]
I'm travelling around this days, will get to review the code in detail again after this Friday. Sorry for delays. The gif looks good 👍

**Comment by mcoffin** [comment]
LGTM

**Comment by farcaller** [comment]
Looks good, sorry for the lag.

**Pull request merged**